### PR TITLE
Add app describe the interaction between students and mentors.

### PR DESCRIPTION
**Pull request by Megonar**

### CODE REVIEW

**Comment by B0RlS** [comment]
Why did you do everything in one commit? I would like you to split everything into several commits

File: HW04/Ramanovich_Dzmitry/lib/homework.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class Homework
+  def initialize(title:, description:, student:, mentor:)
+    @title = title
+    @description = description
+    @student = student
+    @mentor = mentor
+    send_notification_to_student
```
Comment (B0RlS) [inline HW04/Ramanovich_Dzmitry/lib/homework.rb:9]: It shouldn't be here

File: HW04/Ramanovich_Dzmitry/lib/mentor.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @surname = surname
+    @students = []
+    @notifications = []
+  end
+  attr_reader :students
+  attr_accessor :notifications
+
+  def add_homework(title:, description:, student:)
+    homework = Homework.new(title: title, description: description, student: student, mentor: self)
+    student.homeworks += homework
```
Comment (B0RlS) [inline HW04/Ramanovich_Dzmitry/lib/mentor.rb:15]: what if the mentor has a lot of students?

File: HW04/Ramanovich_Dzmitry/lib/mentor.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    homework.student.notifications += notification
+  end
+
+  def accept!(homework)
+    notification = Notification.new(receiver: homework.student, message: 'Homework accepted!')
+    homework.student.notifications += notification
+  end
+
+  def homeworks_to_check
+    notifications_about_homeworks_to_check = notifications.select { |notification| notification.type == :to_check }
```
Comment (B0RlS) [inline HW04/Ramanovich_Dzmitry/lib/mentor.rb:38]: Very strange variable name

File: HW04/Ramanovich_Dzmitry/lib/homework.rb (hunk 1 above)
Comment (Megonar) [reply HW04/Ramanovich_Dzmitry/lib/homework.rb:9]: Moved this method call to mentor#add_homework method

File: HW04/Ramanovich_Dzmitry/lib/mentor.rb (hunk 2 above)
Comment (Megonar) [reply HW04/Ramanovich_Dzmitry/lib/mentor.rb:15]: Done

File: HW04/example.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
```
Comment (B0RlS) [inline HW04/example.rb:2]:
```suggestion
```

File: HW04/app.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
```
Comment (B0RlS) [inline HW04/app.rb:2]:
```suggestion
```

File: HW04/Ramanovich_Dzmitry/lib/student.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    homework.add_answer(answer)
+  end
+
+  def to_check!(homework)
+    if homework.mentor.students.include?(self)
+      notification = Notification.new(receiver: homework.mentor, message: 'New homework to check!',
+                                      subject: homework)
+      homework.mentor.notifications += notification
+    else
+      puts 'To receive notifications from this student, please subscribe!'
```
Comment (B0RlS) [inline HW04/Ramanovich_Dzmitry/lib/student.rb:38]: You have duplicated code in this and 24 line, it is better to put this warning in a private method and call it, and do not duplicate the code.

**Comment by B0RlS** [comment]
fix conflict please

File: HW04/Ramanovich_Dzmitry/lib/student.rb (hunk 6 above)
Comment (Megonar) [reply HW04/Ramanovich_Dzmitry/lib/student.rb]: Done. I created new private method.

File: HW04/Ramanovich_Dzmitry/lib/mentor.rb (hunk 3 above)
Comment (Megonar) [reply HW04/Ramanovich_Dzmitry/lib/mentor.rb]: Done

**Pull request merged**